src/puppetlabs/trapperkeeper/services/webserver/jetty9_core.clj

```diff
@@ -547,6 +560,9 @@
 
 (schema/defn ^:always-validate shutdown
   [webserver-context :- ServerContext]
+  (when-let [{:keys [mbean-container]} @(:state webserver-context)]
+    (log/debug "Cleaning up JMX MBean container")
+    (.destroy mbean-container))
```

Comment:
This seems to be throwing a null pointer exception with `jmx-enabled: "false"` set. It looks like if a key is present in a map, even if the value of that key is falsy, `when-let` will use it. (actually, from reading up on this a bit more, it looks like the `when-let` is completely dependent on the value on the right for the conditional).

Seems like it might be a good idea to have a test of shutting down the server when jmx reporting is disabled too?